src/bytes.rs

```diff
@@ -52,6 +53,23 @@ impl Bytes {
             from_glib_full(glib_ffi::g_bytes_new_static(data.as_ptr() as *const _, data.len()))
         }
     }
+
+    /// Takes ownership of `data` and creates a new `Bytes` without copying.
+    pub fn from_owned<T: AsRef<[u8]> + Send + 'static>(data: T) -> Bytes {
+        let data: Box<Box<AsRef<[u8]>>> = Box::new(Box::new(data));
```

Comment:
Maybe we should use `data: Box<Box<T>>` here?

Reply:
Doesn't really make a difference. We only have the box to keep around the data and free it later when needed. If we would make it a specific box instead of a trait object, a new drop function would have to be created by the compiler for each possibly type passed in here.

Reply:
In current case each `from_owned` will not have own copy of `drop_box`?

Reply:
No, there's only a single function that drops a box that contains a trait object. It's not a generic function

Reply:
Thanks, I thought that it created for any instance of outer generic function

Reply:
Defining a plain function inside another function only makes it invisible from outside the function, it's a way of hiding. Try using any of the outer functions generic types in the inner function, the compiler will complain that it can't do that :)

Same thing with static variables that you define inside a function btw